### DISCUSSION TITLE
Getting "EFBFBD" when pasting from the reader

### DISCUSSION THREAD

**Discussion by rogerbratseth** [sumatrapdfreader/sumatrapdf#3931] [Q&A]
When I print pages from a certain website to PDF, open it in SumatraPDF, and copies text from SumatraPDF to an editor I get the "EFBFBD" on the character combinations "fi" and "fl". Anyone else seen this? I know it can happen when casting char arrays to Strings.
My Windows OS is Norwegian.
The website I'm looking at is: www.learncpp.com
I'm using Firefox v120.0, and the default "Save to PDF" printer.
OS is Windows 10 Pro, 22H2, build 19045.3693

/Roger

**Comment by GitHubRulesOK** [comment]
That is common PDF misbehaviour (getting worse by the years as PDF generation drifts away from the original simplistic concepts).

The PDF writer may convert characters into [combinations called ligatures](https://en.wikipedia.org/wiki/Ligature_(writing)), in a good PDF writer there should be an option to switch that off. MS Word and Edge are both perfect examples of how not to generate PDF with poor encoding by not showing all the options.

Here I save as PDF from the site you mention and the font in use has ligatures but the ligature characters are not conventional text so the single letters are spanned as a different object breaking the string of text.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/d59e0a40-b267-4afa-95f0-e220fdbf57d2)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/f49866ae-3f4d-4fa4-a423-c4f6c80ebdd4)

**Comment by rogerbratseth** [comment]
Hi and thanks for your reply. I think I have a better understanding of this
problem now.

/Roger
